libtenzir/builtins/formats/kv.cpp

```diff
@@ -63,11 +94,33 @@ class splitter {
     TENZIR_ASSERT(regex_);
     TENZIR_ASSERT(regex_->NumberOfCapturingGroups() == 1);
     auto group = re2::StringPiece{};
-    if (not re2::RE2::PartialMatch(input, *regex_, &group)) {
-      return {input, {}};
+    auto start_offset = 0;
+    while (true) {
+      if (not re2::RE2::PartialMatch(input.substr(start_offset), *regex_,
+                                     &group)) {
+        return {input, {}};
+      }
+      if (group.empty()) {
+        return {input, {}};
+      }
```

Comment:
I assume you added this in response to "no infinite loop". You probably want to place this check at the end and use `head.size() + group.size() == 0` instead, it's a bit too strict otherwise.